File: templates/kotlin/src/test/kotlin/CommonTest.kt
// Simple Kotlin test runner (no JUnit dependency)

fun assertEquals(
    expected: Int,
    actual: Int,
    message: String,
) {
  if (expected != actual) {
    throw AssertionError("$message: expected $expected but got $actual")
  }
}

fun runTests() {
  // Add test functions here
  test1()
  test2()

  println("✅ All tests passed!")
}

fun test1() {
  // TODO: Add sample input and expected output from problem description
  val input =
      """
        sample input here
        """
          .trimIndent()
  val expected = 0 // TODO: Update with expected value
  val result = solvePart1(input)
  assertEquals(expected, result, "Part 1 sample test")
}

fun test2() {
  // TODO: Add sample input and expected output from problem description
  val input =
      """
        sample input here
        """
          .trimIndent()
  val expected = 0 // TODO: Update with expected value
  val result = solvePart2(input)
  assertEquals(expected, result, "Part 2 sample test")
}

fun main() {
  runTests()
}

File: src/AoC25/s25e12-go/common/common_test.go
package common

import "testing"

const sample_input string = `0:
###
##.
##.

1:
###
##.
.##

2:
.##
###
##.

3:
##.
###
##.

4:
###
#..
###

5:
###
.#.
###

4x4: 0 0 0 0 2 0
12x5: 1 0 1 0 2 2
12x5: 1 0 1 0 3 2`

func TestSolvePart1Sample1(t *testing.T) {
	result := SolvePart1(sample_input)
	expected := 2
	if result != expected {
		t.Errorf("SolvePart1(%q) = %d; want %d", "", result, expected)
	}
}

func TestSolvePart2Sample1(t *testing.T) {
	result := SolvePart2(sample_input)
	expected := 0 // TODO: Add expected result
	if result != expected {
		t.Errorf("SolvePart2(%q) = %d; want %d", "", result, expected)
	}
}
